### PR TITLE
[BLCK-80] [BIT-96] Get mempool txs wrapper, rename fetch_txns

### PR DESCRIPTION
**Pull request by dario-ramos**

### CODE REVIEW

File: include/bitprim/nodecint/chain/chain.h [hunk 1]
```diff
@@ -148,7 +148,7 @@ BITPRIM_EXPORT
 error_code_t chain_get_history(chain_t chain, payment_address_t address, uint64_t /*size_t*/ limit, uint64_t /*size_t*/ from_height, history_compact_list_t* out_history);
 
 BITPRIM_EXPORT
-void chain_fetch_txns(chain_t chain, void* ctx, payment_address_t address, uint64_t max, uint64_t start_height, transactions_by_addres_fetch_handler_t handler);
+void chain_fetch_confirmed_transactions(chain_t chain, void* ctx, payment_address_t address, uint64_t max, uint64_t start_height, transactions_by_addres_fetch_handler_t handler);
```
Comment (fpelliccioni) [inline include/bitprim/nodecint/chain/chain.h]: Please add chain_get_confirmed_transactions(...)

File: include/bitprim/nodecint/chain/mempool_transaction.h [hunk 2]
```diff
@@ -0,0 +1,58 @@
+/**
```
Comment (fpelliccioni) [inline include/bitprim/nodecint/chain/mempool_transaction.h]: Is this file necessary? Could `transaction` be used?

File: include/bitprim/nodecint/chain/mempool_transaction_list.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/**
```
Comment (fpelliccioni) [inline include/bitprim/nodecint/chain/mempool_transaction_list.h]: Could `transaction_list` be used?

File: src/chain/chain.cpp [hunk 4]
```diff
@@ -760,6 +760,19 @@ error_code_t chain_get_stealth(chain_t chain, void* ctx, binary_t filter, uint64
 //virtual void fetch_template(merkle_block_fetch_handler handler) const = 0;
 //virtual void fetch_mempool(size_t count_limit, uint64_t minimum_fee, inventory_fetch_handler handler) const = 0;
 //
+
+void chain_get_mempool_transactions(chain_t chain, payment_address_t address, int /*bool*/ use_testnet_rules, int /*bool*/ witness, mempool_transaction_list_t* out_txs) {
```
Comment (fpelliccioni) [inline src/chain/chain.cpp]: Use return value instead of output parameters...

File: include/bitprim/nodecint/chain/chain.h [hunk 5]
```diff
@@ -182,7 +182,7 @@ error_code_t chain_get_stealth(chain_t chain, void* ctx, binary_t filter, uint64
 //virtual void fetch_template(merkle_block_fetch_handler handler) const = 0;
 //virtual void fetch_mempool(size_t count_limit, uint64_t minimum_fee, inventory_fetch_handler handler) const = 0;
 
-void get_mempool_transactions(chain_t chain, payment_address_t address, int /*bool*/ use_testnet_rules, int /*bool*/ witness, mempool_transaction_list_t* out_txs);
+void chain_get_mempool_transactions(chain_t chain, payment_address_t address, int /*bool*/ use_testnet_rules, int /*bool*/ witness, mempool_transaction_list_t* out_txs);
```
Comment (fpelliccioni) [inline include/bitprim/nodecint/chain/chain.h]: Use return value instead of output parameters.

File: include/bitprim/nodecint/chain/chain.h [hunk 6]
```diff
@@ -182,7 +182,7 @@ error_code_t chain_get_stealth(chain_t chain, void* ctx, binary_t filter, uint64
...
```
Comment (fpelliccioni) [inline include/bitprim/nodecint/chain/chain.h]: Add the chain_fetch_... variant

Comment (dario-ramos) [reply include/bitprim/nodecint/chain/chain.h]: It doesn't make sense to implement an async version, because these values are read straight from memory, there is no I/O involved

File: include/bitprim/nodecint/chain/mempool_transaction.h (hunk 2 above)
Comment (dario-ramos) [reply include/bitprim/nodecint/chain/mempool_transaction.h]: Registered in Jira as BIT-106
